packages/ketcher-core/src/chemistry/entities/SGroup.ts

```diff
@@ -187,32 +187,39 @@ export class SGroup {
   }
 
   static getCrossBonds(
-    inBonds: any,
-    xBonds: any,
     mol: any,
     parentAtomSet: Pile<number>
-  ): void {
+  ): { [key: number]: Array<Bond> } {
+    const crossBonds: { [key: number]: Array<Bond> }= {}
     mol.bonds.forEach((bond, bid) => {
-      if (parentAtomSet.has(bond.begin) && parentAtomSet.has(bond.end)) {
-        if (inBonds !== null) inBonds.push(bid)
-      } else if (parentAtomSet.has(bond.begin) || parentAtomSet.has(bond.end)) {
-        if (xBonds !== null) xBonds.push(bid)
+      if (parentAtomSet.has(bond.begin) && !parentAtomSet.has(bond.end)) {
+        if (!crossBonds[bond.begin]) {
+          crossBonds[bond.begin] = []
+        }
+        crossBonds[bond.begin].push(bid)
+      } else if (parentAtomSet.has(bond.end) && !parentAtomSet.has(bond.begin) ) {
+        if (!crossBonds[bond.end]) {
+          crossBonds[bond.end] = []
+        }
+        crossBonds[bond.end].push(bid)
       }
     })
+    return crossBonds
   }
 
-  static bracketPos(sg, mol, xbonds): void {
-    var atoms = sg.atoms
-    if (!xbonds || xbonds.length !== 2) {
-      sg.bracketDir = new Vec2(1, 0)
+  static bracketPos(sGroup, mol, crossBondsPerAtom: { [key: number]: Array<Bond> }): void {
+    var atoms = sGroup.atoms
+    const crossBonds = Array.prototype.concat.apply([], Object.values(crossBondsPerAtom))
```

Comment:
please simplify 
`Array.prototype.concat.apply([], Object.values(crossBondsPerAtom))`
to 
`Object.values(crossBondsPerAtom).flat()`
or
`[].concat(...Object.values(crossBondsPerAtom))`